analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -254,6 +254,44 @@ public void Conversion_CustomOperators_DoNotPropagateState_Half()
 
 #endif
 
+        [TestMethod]
+        public void Conversion_ToLocalNonNullableValueType_CS()
+        {
+            var validator = SETestContext.CreateCS("""
+                object o = null; // Set ObjectConstraint.Null and Dummy (LiteralDummyTestCheck)
+                int convert = System.Convert.ToInt32(o);
+                int explicitCast = (int)o;
+                Tag("Object", o);
+                Tag("Convert", convert);
+                Tag("Explicit", explicitCast);
+                """, new LiteralDummyTestCheck()).Validator;
+            validator.ValidateTag("Object", x => x.Should().HaveOnlyConstraints(ObjectConstraint.Null, DummyConstraint.Dummy));
+            validator.ValidateTag("Convert", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull));
+            validator.ValidateTag("Explicit", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, DummyConstraint.Dummy));
+        }
+
+        [TestMethod]
+        public void Conversion_ToLocalNonNullableValueType_VB()
+        {
+            var validator = SETestContext.CreateVB("""
+                Dim o As Object = Nothing ' Set ObjectConstraint.Null and Dummy (LiteralDummyTestCheck)
+                Dim iCType As Integer
+                Dim iDirectCast As Integer
+                Dim iImplicit As Integer
+                iCType = CType(o, Integer)
```

Comment:
Let's add also `iCInt = CInt(o)` to see how it behaves. It should behave the same as CType in the best case scenario